docs/guides/migration.md

```diff
@@ -0,0 +1,223 @@
+---
+subcategory: "Guides"
+page_title: "Migration"
+description: |-
+    An overview of resource and datasource migration changes
```

Comment:
Can we add migration from Terraform SDKv2 to Plugin Framework?

Reply:
or just migration to the latest Terraform Plugin Framework?

Reply:
removed latest but went with your second suggestion and also added to title